staging/src/k8s.io/cli-runtime/pkg/resource/selector.go

```diff
@@ -81,6 +90,69 @@ func (r *Selector) Visit(fn VisitorFunc) error {
 	})
 }
 
+func (r *Selector) watchList(fn VisitorFunc, listOptions metav1.ListOptions, helper *Helper) error {
```

Comment:
This mechanism is only beta in 1.31, whereas kubectl ensures +/-1 compatibility. I'm not saying we should not enable this functionality, very likely due to its performance gains we'll switch to it by default, but only after this feature graduates to GA, which most likely be not sooner than 1.33, although looking at https://github.com/kubernetes/enhancements/issues/3157 it's still TBD.

Reply:
yes, I totally agree with that. But in currently stage, do we have to wait for WatchList being Beta/GA, or we can put this into Alpha and disabled by default?

Reply:
Since this will potentially have a high impact on users, it should follow the usual enhancement process, where we we will ensure backwards compatibility and a proper fallback path, when this new mechanism is not available.

Reply:
Sure, I will submit a KEP ASAP.